### ISSUE TITLE
When I killed Infernum supureme calamitas the game crash everytime.

### ISSUE THREAD

**Issue by hamagon** [DominicKarma/InfernumModePublic#66]
when I killed supureme calamitas in Infernum mode, the game crash.

error message is  " failed to load asset "Content¥Credits¥Textures¥Joey"!  "
it also say that "Main engine crash"
I did game file integrity check on steam but it doesn't solve it.

![スクリーンショット (1228)](https://github.com/DominicKarma/InfernumModePublic/assets/96931214/4f2174be-8a77-4754-a7bd-e98a1710137e)
![スクリーンショット (1229)](https://github.com/DominicKarma/InfernumModePublic/assets/96931214/726a5cd1-78d1-4dd3-a6f1-5ed90d585db2)

**Comment by emeraldviridis** [comment]
I am experiencing the same issue. Playing with tModLoader 1.4.3 on a world that's about 2 months old. Tried reinstalling Infernum and verifying integrity too.

**Comment by Cryptic82501** [comment]
I got the same issue 1.4.3 with a 2 or 3 month old world. Tried both reinstalling and verifying.

**Comment by Maxcheniour** [comment]
I've encountered the same problem on a month old world, here is the entire error log from my game:
`[16:57:38.018] [Main Thread/WARN] [tML]: Silently Caught Exception: 
ReLogic.Content.AssetLoadException: Asset could not be found: "Content\Credits\Textures\Joey"
   at ReLogic.Content.AssetRepository.LoadAssetWithPotentialAsync[T](Asset`1 asset, AssetRequestMode mode) in D:\a\tModLoader\tModLoader\src\tModLoader\ReLogic\Content\AssetRepository.cs:line 262
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine& stateMachine)
   at ReLogic.Content.AssetRepository.LoadAssetWithPotentialAsync[T](Asset`1 asset, AssetRequestMode mode)
   at ReLogic.Content.AssetRepository.LoadAsset[T](Asset`1 asset, AssetRequestMode mode) in D:\a\tModLoader\tModLoader\src\tModLoader\ReLogic\Content\AssetRepository.cs:line 209
   at ReLogic.Content.AssetRepository.Request[T](String assetName, AssetRequestMode mode) in D:\a\tModLoader\tModLoader\src\tModLoader\ReLogic\Content\AssetRepository.cs:line 164
   at Terraria.ModLoader.ModContent.Request[T](String name, AssetRequestMode mode) in tModLoader\Terraria\ModLoader\ModContent.cs:line 116
   at InfernumMode.Content.Credits.CreditFinalScene.SetupObjects() in InfernumMode\Content\Credits\CreditFinalScene.cs:line 56
   at InfernumMode.Content.Credits.CreditManager.BeginCredits() in InfernumMode\Content\Credits\CreditManager.cs:line 142
   at InfernumMode.Content.BehaviorOverrides.BossAIs.SupremeCalamitas.SupremeCalamitasBehaviorOverride.DoBehavior_DesperationPhase(NPC npc, Player target, Single& frameType, Single& frameChangeSpeed, Single& attackTimer) in InfernumMode\Content\BehaviorOverrides\BossAIs\SupremeCalamitas\SupremeCalamitasBehaviorOverride.cs:line 2325
   at InfernumMode.Content.BehaviorOverrides.BossAIs.SupremeCalamitas.SupremeCalamitasBehaviorOverride.PreAI(NPC npc) in InfernumMode\Content\BehaviorOverrides\BossAIs\SupremeCalamitas\SupremeCalamitasBehaviorOverride.cs:line 478
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Span`1& arguments, Signature sig, Boolean constructor, Boolean wrapExceptions)
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at System.Reflection.MethodBase.Invoke(Object obj, Object[] parameters)
   at InfernumMode.Core.OverridingSystem.NPCBehaviorOverride.<>c__DisplayClass1_0.<LoadAll>b__0(NPC n) in InfernumMode\Core\OverridingSystem\NPCBehaviorOverride.cs:line 29
   at InfernumMode.Core.GlobalInstances.GlobalNPCOverrides.PreAI(NPC npc) in InfernumMode\Core\GlobalInstances\GlobalNPCOverrides.cs:line 193
   at InfernumMode.Core.ILEditingStuff.OverrideSystemHooks.<>c.<NPCPreAIChange>b__0_0(NPC npc) in InfernumMode\Core\ILEditingStuff\OverrideSystemHooks.cs:line 42
   at DMD<Terraria.ModLoader.NPCLoader::PreAI>(NPC npc)
   at Terraria.ModLoader.NPCLoader.NPCAI(NPC npc) in tModLoader\Terraria\ModLoader\NPCLoader.cs:line 235
   at Terraria.NPC.UpdateNPC_Inner(Int32 i) in tModLoader\Terraria\NPC.cs:line 69393
   at DMD<Terraria.Main::DoUpdateInWorld>(Main this, Stopwatch sw)
   at Terraria.Main.DoUpdate(GameTime& gameTime) in tModLoader\Terraria\Main.cs:line 13223
   at Terraria.Main.Update(GameTime gameTime) in tModLoader\Terraria\Main.cs:line 12778
   at Microsoft.Xna.Framework.Game.Tick() in D:\a\tModLoader\tModLoader\FNA\src\Game.cs:line 546
   at Microsoft.Xna.Framework.Game.RunLoop() in D:\a\tModLoader\tModLoader\FNA\src\Game.cs:line 878
   at Microsoft.Xna.Framework.Game.Run() in D:\a\tModLoader\tModLoader\FNA\src\Game.cs:line 419
   at Terraria.Program.LaunchGame_(Boolean isServer) in tModLoader\Terraria\Program.cs:line 227
   at Terraria.Program.LaunchGame(String[] args, Boolean monoArgs) in tModLoader\Terraria\Program.cs:line 188
   at MonoLaunch.<>c__DisplayClass1_0.<Main>b__0() in tModLoader\Terraria\MonoLaunch.cs:line 56
   at System.Threading.Thread.StartCallback()

[16:57:38.022] [Main Thread/ERROR] [Terraria]: Failed to load asset: "Content\Credits\Textures\Joey"
ReLogic.Content.AssetLoadException: Asset could not be found: "Content\Credits\Textures\Joey"
   at ReLogic.Content.AssetRepository.LoadAssetWithPotentialAsync[T](Asset`1 asset, AssetRequestMode mode) in D:\a\tModLoader\tModLoader\src\tModLoader\ReLogic\Content\AssetRepository.cs:line 262
[16:57:51.435] [Main Thread/WARN] [tML]: Silently Caught Exception: 
System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation.
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Span`1& arguments, Signature sig, Boolean constructor, Boolean wrapExceptions)
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at System.Reflection.MethodBase.Invoke(Object obj, Object[] parameters)
   at InfernumMode.Core.OverridingSystem.NPCBehaviorOverride.<>c__DisplayClass1_0.<LoadAll>b__0(NPC n) in InfernumMode\Core\OverridingSystem\NPCBehaviorOverride.cs:line 29
   at InfernumMode.Core.GlobalInstances.GlobalNPCOverrides.PreAI(NPC npc) in InfernumMode\Core\GlobalInstances\GlobalNPCOverrides.cs:line 193
   at InfernumMode.Core.ILEditingStuff.OverrideSystemHooks.<>c.<NPCPreAIChange>b__0_0(NPC npc) in InfernumMode\Core\ILEditingStuff\OverrideSystemHooks.cs:line 42
   at DMD<Terraria.ModLoader.NPCLoader::PreAI>(NPC npc)
   at Terraria.ModLoader.NPCLoader.NPCAI(NPC npc) in tModLoader\Terraria\ModLoader\NPCLoader.cs:line 235
   at Terraria.NPC.UpdateNPC_Inner(Int32 i) in tModLoader\Terraria\NPC.cs:line 69393
   at DMD<Terraria.Main::DoUpdateInWorld>(Main this, Stopwatch sw)
   at Terraria.Main.DoUpdate(GameTime& gameTime) in tModLoader\Terraria\Main.cs:line 13223
   at Terraria.Main.Update(GameTime gameTime) in tModLoader\Terraria\Main.cs:line 12778
   at Microsoft.Xna.Framework.Game.Tick() in D:\a\tModLoader\tModLoader\FNA\src\Game.cs:line 546
   at Microsoft.Xna.Framework.Game.RunLoop() in D:\a\tModLoader\tModLoader\FNA\src\Game.cs:line 878
   at Microsoft.Xna.Framework.Game.Run() in D:\a\tModLoader\tModLoader\FNA\src\Game.cs:line 419
   at Terraria.Program.LaunchGame_(Boolean isServer) in tModLoader\Terraria\Program.cs:line 227
   at Terraria.Program.LaunchGame(String[] args, Boolean monoArgs) in tModLoader\Terraria\Program.cs:line 188
   at MonoLaunch.<>c__DisplayClass1_0.<Main>b__0() in tModLoader\Terraria\MonoLaunch.cs:line 56
   at System.Threading.Thread.StartCallback()

[16:57:51.455] [Main Thread/WARN] [tML]: Silently Caught Exception: 
System.Collections.Generic.KeyNotFoundException: Instance of 'GlobalNPCOverrides' does not exist on the current NPC.
   at Terraria.ModLoader.GlobalType.GetGlobal[TEntity,TGlobal,TResult](Instanced`1[] globals, Boolean exactType) in tModLoader\Terraria\ModLoader\GlobalType.cs:line 49
   at InfernumMode.Utilities.Infernum(NPC npc) in InfernumMode\Common\UtilityMethods\GlobalExtensions.cs:line 38
   at InfernumMode.Content.Skies.SCalScreenShaderData.Apply() in InfernumMode\Content\Skies\SCalScreenShaderData.cs:line 57
   at DMD<Terraria.Graphics.Effects.FilterManager::EndCapture>(FilterManager this, RenderTarget2D finalTexture, RenderTarget2D screenTarget1, RenderTarget2D screenTarget2, Color clearColor)
   at Trampoline<Terraria.Graphics.Effects.FilterManager::EndCapture>?13994550(FilterManager , RenderTarget2D , RenderTarget2D , RenderTarget2D , Color )
   at InfernumMode.Common.Graphics.ScreenEffects.FilterCaputureDetourManager.EndCaptureManager(orig_EndCapture orig, FilterManager self, RenderTarget2D finalTexture, RenderTarget2D screenTarget1, RenderTarget2D screenTarget2, Color clearColor) in InfernumMode\Common\Graphics\ScreenEffects\FilterCaputureDetourManager.cs:line 26
   at Hook<Terraria.Graphics.Effects.FilterManager::EndCapture>?10892990(FilterManager , RenderTarget2D , RenderTarget2D , RenderTarget2D , Color )
   at Trampoline<Terraria.Graphics.Effects.FilterManager::EndCapture>?8746788(FilterManager , RenderTarget2D , RenderTarget2D , RenderTarget2D , Color )
   at InfernumMode.Common.Graphics.ScreenEffects.ScreenSaturationBlurSystem.GetFinalScreenShader(orig_EndCapture orig, FilterManager self, RenderTarget2D finalTexture, RenderTarget2D screenTarget1, RenderTarget2D screenTarget2, Color clearColor) in InfernumMode\Common\Graphics\ScreenEffects\ScreenSaturationBlurSystem.cs:line 168
   at Hook<Terraria.Graphics.Effects.FilterManager::EndCapture>?14254270(FilterManager , RenderTarget2D , RenderTarget2D , RenderTarget2D , Color )
   at Trampoline<Terraria.Graphics.Effects.FilterManager::EndCapture>?66383444(FilterManager , RenderTarget2D , RenderTarget2D , RenderTarget2D , Color )
   at Lights.Lights.FilterManager_EndCapture(orig_EndCapture orig, FilterManager self, RenderTarget2D finalTexture, RenderTarget2D screenTarget1, RenderTarget2D screenTarget2, Color clearColor) in Lights.cs:line 81
   at Hook<Terraria.Graphics.Effects.FilterManager::EndCapture>?42357222(FilterManager , RenderTarget2D , RenderTarget2D , RenderTarget2D , Color )
   at DMD<Terraria.Main::DoDraw>(Main this, GameTime gameTime)
   at Terraria.Main.Draw(GameTime gameTime) in tModLoader\Terraria\Main.cs:line 48713
   at Microsoft.Xna.Framework.Game.Tick() in D:\a\tModLoader\tModLoader\FNA\src\Game.cs:line 562
   at Microsoft.Xna.Framework.Game.RunLoop() in D:\a\tModLoader\tModLoader\FNA\src\Game.cs:line 878
   at Microsoft.Xna.Framework.Game.Run() in D:\a\tModLoader\tModLoader\FNA\src\Game.cs:line 419
   at Terraria.Program.LaunchGame_(Boolean isServer) in tModLoader\Terraria\Program.cs:line 227
   at Terraria.Program.LaunchGame(String[] args, Boolean monoArgs) in tModLoader\Terraria\Program.cs:line 188
   at MonoLaunch.<>c__DisplayClass1_0.<Main>b__0() in tModLoader\Terraria\MonoLaunch.cs:line 56
   at System.Threading.Thread.StartCallback()

[16:57:51.457] [Main Thread/ERROR] [Terraria]: System.Collections.Generic.KeyNotFoundException: Instance of 'GlobalNPCOverrides' does not exist on the current NPC.
   at Terraria.ModLoader.GlobalType.GetGlobal[TEntity,TGlobal,TResult](Instanced`1[] globals, Boolean exactType) in tModLoader\Terraria\ModLoader\GlobalType.cs:line 49
   at InfernumMode.Utilities.Infernum(NPC npc) in InfernumMode\Common\UtilityMethods\GlobalExtensions.cs:line 38
   at InfernumMode.Content.Skies.SCalScreenShaderData.Apply() in InfernumMode\Content\Skies\SCalScreenShaderData.cs:line 57
   at DMD<Terraria.Graphics.Effects.FilterManager::EndCapture>(FilterManager this, RenderTarget2D finalTexture, RenderTarget2D screenTarget1, RenderTarget2D screenTarget2, Color clearColor)
   at Trampoline<Terraria.Graphics.Effects.FilterManager::EndCapture>?13994550(FilterManager , RenderTarget2D , RenderTarget2D , RenderTarget2D , Color )
   at InfernumMode.Common.Graphics.ScreenEffects.FilterCaputureDetourManager.EndCaptureManager(orig_EndCapture orig, FilterManager self, RenderTarget2D finalTexture, RenderTarget2D screenTarget1, RenderTarget2D screenTarget2, Color clearColor) in InfernumMode\Common\Graphics\ScreenEffects\FilterCaputureDetourManager.cs:line 26
   at Hook<Terraria.Graphics.Effects.FilterManager::EndCapture>?10892990(FilterManager , RenderTarget2D , RenderTarget2D , RenderTarget2D , Color )
   at Trampoline<Terraria.Graphics.Effects.FilterManager::EndCapture>?8746788(FilterManager , RenderTarget2D , RenderTarget2D , RenderTarget2D , Color )
   at InfernumMode.Common.Graphics.ScreenEffects.ScreenSaturationBlurSystem.GetFinalScreenShader(orig_EndCapture orig, FilterManager self, RenderTarget2D finalTexture, RenderTarget2D screenTarget1, RenderTarget2D screenTarget2, Color clearColor) in InfernumMode\Common\Graphics\ScreenEffects\ScreenSaturationBlurSystem.cs:line 168
   at Hook<Terraria.Graphics.Effects.FilterManager::EndCapture>?14254270(FilterManager , RenderTarget2D , RenderTarget2D , RenderTarget2D , Color )
   at Trampoline<Terraria.Graphics.Effects.FilterManager::EndCapture>?66383444(FilterManager , RenderTarget2D , RenderTarget2D , RenderTarget2D , Color )
   at Lights.Lights.FilterManager_EndCapture(orig_EndCapture orig, FilterManager self, RenderTarget2D finalTexture, RenderTarget2D screenTarget1, RenderTarget2D screenTarget2, Color clearColor) in Lights.cs:line 81
   at Hook<Terraria.Graphics.Effects.FilterManager::EndCapture>?42357222(FilterManager , RenderTarget2D , RenderTarget2D , RenderTarget2D , Color )
   at DMD<Terraria.Main::DoDraw>(Main this, GameTime gameTime)
   at Terraria.Main.Draw(GameTime gameTime) in tModLoader\Terraria\Main.cs:line 48724
[16:57:51.517] [Main Thread/FATAL] [tML]: Main engine crash

System.Collections.Generic.KeyNotFoundException: Instance of 'GlobalNPCOverrides' does not exist on the current NPC.
   at Terraria.ModLoader.GlobalType.GetGlobal[TEntity,TGlobal,TResult](Instanced`1[] globals, Boolean exactType) in tModLoader\Terraria\ModLoader\GlobalType.cs:line 49
   at InfernumMode.Utilities.Infernum(NPC npc) in InfernumMode\Common\UtilityMethods\GlobalExtensions.cs:line 38
   at InfernumMode.Content.Skies.SCalScreenShaderData.Apply() in InfernumMode\Content\Skies\SCalScreenShaderData.cs:line 57
   at DMD<Terraria.Graphics.Effects.FilterManager::EndCapture>(FilterManager this, RenderTarget2D finalTexture, RenderTarget2D screenTarget1, RenderTarget2D screenTarget2, Color clearColor)
   at Trampoline<Terraria.Graphics.Effects.FilterManager::EndCapture>?13994550(FilterManager , RenderTarget2D , RenderTarget2D , RenderTarget2D , Color )
   at InfernumMode.Common.Graphics.ScreenEffects.FilterCaputureDetourManager.EndCaptureManager(orig_EndCapture orig, FilterManager self, RenderTarget2D finalTexture, RenderTarget2D screenTarget1, RenderTarget2D screenTarget2, Color clearColor) in InfernumMode\Common\Graphics\ScreenEffects\FilterCaputureDetourManager.cs:line 26
   at Hook<Terraria.Graphics.Effects.FilterManager::EndCapture>?10892990(FilterManager , RenderTarget2D , RenderTarget2D , RenderTarget2D , Color )
   at Trampoline<Terraria.Graphics.Effects.FilterManager::EndCapture>?8746788(FilterManager , RenderTarget2D , RenderTarget2D , RenderTarget2D , Color )
   at InfernumMode.Common.Graphics.ScreenEffects.ScreenSaturationBlurSystem.GetFinalScreenShader(orig_EndCapture orig, FilterManager self, RenderTarget2D finalTexture, RenderTarget2D screenTarget1, RenderTarget2D screenTarget2, Color clearColor) in InfernumMode\Common\Graphics\ScreenEffects\ScreenSaturationBlurSystem.cs:line 168
   at Hook<Terraria.Graphics.Effects.FilterManager::EndCapture>?14254270(FilterManager , RenderTarget2D , RenderTarget2D , RenderTarget2D , Color )
   at Trampoline<Terraria.Graphics.Effects.FilterManager::EndCapture>?66383444(FilterManager , RenderTarget2D , RenderTarget2D , RenderTarget2D , Color )
   at Lights.Lights.FilterManager_EndCapture(orig_EndCapture orig, FilterManager self, RenderTarget2D finalTexture, RenderTarget2D screenTarget1, RenderTarget2D screenTarget2, Color clearColor) in Lights.cs:line 81
   at Hook<Terraria.Graphics.Effects.FilterManager::EndCapture>?42357222(FilterManager , RenderTarget2D , RenderTarget2D , RenderTarget2D , Color )
   at DMD<Terraria.Main::DoDraw>(Main this, GameTime gameTime)
   at Terraria.Main.Draw(GameTime gameTime) in tModLoader\Terraria\Main.cs:line 48724
   at Microsoft.Xna.Framework.Game.Tick() in D:\a\tModLoader\tModLoader\FNA\src\Game.cs:line 562
   at Microsoft.Xna.Framework.Game.RunLoop() in D:\a\tModLoader\tModLoader\FNA\src\Game.cs:line 878
   at Microsoft.Xna.Framework.Game.Run() in D:\a\tModLoader\tModLoader\FNA\src\Game.cs:line 419
   at Terraria.Program.LaunchGame_(Boolean isServer) in tModLoader\Terraria\Program.cs:line 227
[16:57:51.521] [Main Thread/DEBUG] [TerrariaSteamClient]: Send: shutdown`

here is a list of all the mods I used:

`[16:18:58.367] [.NET ThreadPool Worker/INFO] [tML]: Finalizing Content: InfernumModeMusic (Infernum Mode Music) v1.9.15
[16:18:58.498] [.NET ThreadPool Worker/INFO] [tML]: Finalizing Content: InstantHousing (Instant Housing) v1.2.3
[16:18:58.629] [.NET ThreadPool Worker/INFO] [tML]: Finalizing Content: InstantPlatformFallthrough (Instant Platform Fallthrough) v1.1
[16:18:58.759] [.NET ThreadPool Worker/INFO] [tML]: Finalizing Content: Lights (Lights And Shadow) v1.2.2
[16:18:58.892] [.NET ThreadPool Worker/INFO] [tML]: Finalizing Content: SerousCommonLib (absoluteAquarian Utilities) v1.0.4.1
[16:18:59.023] [.NET ThreadPool Worker/INFO] [tML]: Finalizing Content: MagicStorage (Magic Storage) v0.6.0.3
[16:18:59.169] [.NET ThreadPool Worker/INFO] [tML]: Finalizing Content: miningcracks_take_on_luiafk (LuiAFK Reborn) v1.1.5.3
[16:18:59.304] [.NET ThreadPool Worker/INFO] [tML]: Finalizing Content: NoPylonRestrictions (No Pylon Restrictions) v0.1.3.4
[16:18:59.437] [.NET ThreadPool Worker/INFO] [tML]: Finalizing Content: OmniSwing (OmniSwing) v3.1.0
[16:18:59.570] [.NET ThreadPool Worker/INFO] [tML]: Finalizing Content: OreExcavator (Ore Excavator) v0.8.4
[16:18:59.571] [.NET ThreadPool Worker/INFO] [OreExcavator]: Detecting and converting old config entries...
[16:18:59.571] [.NET ThreadPool Worker/INFO] [OreExcavator]: All existing entries are up-to-date!
[16:18:59.571] [.NET ThreadPool Worker/INFO] [OreExcavator]: Auto-whitelist runner started. Looking for modded ores and gems to whitelist...
[16:18:59.571] [.NET ThreadPool Worker/INFO] [OreExcavator]: Running through tML Ore Tile Set...
[16:18:59.571] [.NET ThreadPool Worker/INFO] [OreExcavator]: Ore Set Search completed.
[16:18:59.572] [.NET ThreadPool Worker/INFO] [OreExcavator]: Beginning manaual ore search...
[16:18:59.572] [.NET ThreadPool Worker/INFO] [OreExcavator]: Found CalamityMod - But we're already up-to-date with them! v(2.0.2.3) 
[16:18:59.575] [.NET ThreadPool Worker/INFO] [OreExcavator]: Auto-whitelist runner concluded. Finishing up...
[16:18:59.721] [.NET ThreadPool Worker/INFO] [tML]: Finalizing Content: RecipeBrowser (Recipe Browser) v0.9.9
[16:18:59.853] [.NET ThreadPool Worker/INFO] [tML]: Finalizing Content: StructureHelper (Structure Helper) v2.2.0
[16:18:59.985] [.NET ThreadPool Worker/INFO] [tML]: Finalizing Content: SummonsUI (Summoner UI) v1.1
[16:19:00.121] [.NET ThreadPool Worker/INFO] [tML]: Finalizing Content: UnCalamityModMusic (Vanilla Calamity Mod Music) v1.0.9
[16:19:00.256] [.NET ThreadPool Worker/INFO] [tML]: Finalizing Content: WeaponDisplay (ArmamentDisplay) v1.13
[16:19:00.389] [.NET ThreadPool Worker/INFO] [tML]: Finalizing Content: WingSlotExtra (Wing Slot Extra) v1.3.4
[16:19:00.523] [.NET ThreadPool Worker/INFO] [tML]: Finalizing Content: WombatQOL (Wombat's General Improvements) v1.2.3
[16:19:00.656] [.NET ThreadPool Worker/INFO] [tML]: Finalizing Content: WorldGenPreviewer (WorldGen Previewer) v0.6`
